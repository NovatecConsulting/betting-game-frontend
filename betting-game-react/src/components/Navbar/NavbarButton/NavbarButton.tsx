import React from "react";
import { NavLink } from "react-router-dom";

export interface NavbarButtonProps {
    buttonText: string;
    buttonURL: string;
    buttonTestIdPrefix: string;
}

const NavbarButton: React.FC<NavbarButtonProps> = ({
    buttonText,
    buttonURL,
    buttonTestIdPrefix,
}: NavbarButtonProps) => (
    <NavLink
        to={buttonURL}
        className="mr-4 lg:inline-block lg:mt-0 py-2 border-b-2 border-transparent inline-block text-gray-400 hover:text-white"
        activeClassName="border-gray-400 text-indigo-100"
        data-testid={`${buttonTestIdPrefix}-NavButton`}
    >
        <span data-testid={`${buttonTestIdPrefix}-NavButtonText`}>
            {buttonText}
        </span>
    </NavLink>
);

export default NavbarButton;
