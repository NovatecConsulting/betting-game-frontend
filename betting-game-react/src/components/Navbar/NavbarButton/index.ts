export { default as NavbarButton } from "./NavbarButton";
