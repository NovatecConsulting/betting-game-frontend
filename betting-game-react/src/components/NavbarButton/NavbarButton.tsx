import React from 'react';
import { NavLink } from 'react-router-dom';

export interface NavbarButtonProps {
  buttonText: string;
  buttonURL: string;
  buttonTestID: string;
}

const NavbarButton: React.FC<NavbarButtonProps> = ({
  buttonText,
  buttonURL,
  buttonTestID,
}: NavbarButtonProps) => (
  <NavLink
    to={buttonURL}
    className='block mr-4 lg:inline-block lg:mt-0 py-2 border-b-2 border-transparent inline-block text-gray-400 hover:text-white'
    activeClassName='border-gray-400 text-indigo-100'
    data-testid={buttonTestID}
  >
    {buttonText}
  </NavLink>
);

export default NavbarButton;
