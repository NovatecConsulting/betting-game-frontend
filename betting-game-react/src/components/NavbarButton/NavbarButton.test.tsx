import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import NavbarButton from './NavbarButton';

describe('<NavbarButton />', () => {
  test('it should mount', () => {
    render(<NavbarButton buttonText='' buttonURL='' buttonTestID='' />);

    const navbarButton = screen.getByTestId('NavbarButton');

    expect(navbarButton).toBeInTheDocument();
  });
});
