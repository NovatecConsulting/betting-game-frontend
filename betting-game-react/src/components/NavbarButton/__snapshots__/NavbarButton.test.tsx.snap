// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarButton component should render and match snapshot with correct content. 1`] = `
<NavLink
  activeClassName="border-gray-400 text-indigo-100"
  className="mr-4 lg:inline-block lg:mt-0 py-2 border-b-2 border-transparent inline-block text-gray-400 hover:text-white"
  data-testid="TestButton-NavButton"
  to="/testButtonURL"
>
  <span
    data-testid="TestButton-NavButtonText"
  >
    TestButtonText
  </span>
</NavLink>
`;
